### PR TITLE
New feature: Program creation from the user interface

### PR DESCRIPTION
**Pull request by leirbag95**
This update concerns the creation and import of programs directly from the user interface account.aleph.im.
The Aleph-VM project allows you to run programs on [Aleph.im](https://aleph.im/).

Aleph-VM is optimized to run programs on demand in a "function-as-as-service", as a response to HTTP requests.

What has been done:

+ Add a new section to the menu: Programs.
+ Retrieve the source code of a program
+ Retrieve vms from an added CallToAction, which avoids reloading the whole dapp at each new deployment.
+ Import a python script from an online text editor
+ VM configuration (with or without volume, persistent or not)
+ Signing messages with Metamask

### CODE REVIEW

File: src/components/CreateNewVM.vue [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        }
+        message.item_hash = hash
+      }
+    },
+    async importCode () {
+      this.loading = true
+      const zip = new JSZip()
+      zip.file('main.py', this.newProgram.code)
+      this.exportFile = await zip.generateAsync({ type: 'blob' })
+      this.step = 2
```
Comment (hoh) [inline src/components/CreateNewVM.vue:226]: Would it help if these steps were static variables ?
```javascript
STEP_EDITOR = 1
STEP_DONE = 2
```

File: src/components/CreateNewVM.vue [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+      this.loading = false
+    },
+    async send (message) {
+      if (this.account.type === 'NULS') {
+        message = await nuls.sign(Buffer.from(this.account.private_key, 'hex'), message)
+      } else if (this.account.type === 'NULS2') {
+        message = await nuls2.sign(this.account, message)
+      } else if (this.account.type === 'NEO') {
+        message = await neo.sign(this.account, message)
+      } else if (this.account.type === 'ETH') {
```
Comment (hoh) [inline src/components/CreateNewVM.vue:236]: Should ETH be first since it is the most common case ?

File: src/components/CreateNewVM.vue [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+          chain: 'ETH',
+          sender: this.account.address,
+          account: this.account,
+          channel: 'TEST',
+          fileobject: this.exportFile,
+          api_server: this.api_server
+        }
+      ).catch((response) => {
+        this.loading = false
+      })
```
Comment (hoh) [inline src/components/CreateNewVM.vue:256]: What happends when the backend is not available ?

File: src/pages/VMView.vue [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    async getMessages () {
+      this.loading = true
+      await messages.get_messages({
+        addresses: [this.account.address],
+        pagination: 1000,
+        message_type: 'PROGRAM'
+      }).then(async (response) => {
+        var programsTmp = response.messages
+        for (var i = 0; i < programsTmp.length; i++) {
```
Comment (hoh) [inline src/pages/VMView.vue:81]: Why not `for...of` ?

**Review by hoh — COMMENTED** [review]
This looks very good ! 
I added a few comments in the code.

File: src/components/CreateNewVM.vue (hunk 3 above)
Comment (leirbag95) [reply src/components/CreateNewVM.vue:256]: The 'Deploy' button will stop loading and reactivate again.

File: src/pages/VMView.vue (hunk 4 above)
Comment (leirbag95) [reply src/pages/VMView.vue]: With a `forEach` or a `for...of` I can't modify the content of the object passed with asynchronous way in a props which forces me to reload the display for showing size element.
So I use a classic `for` loop and it works pretty well.

Then I modify the object in the following way
```
programsTmp[i].storeObj = storeObj
```

**Comment by hoh** [comment]
Replaced by #28

**Pull request closed without merging**